core/src/main/scala/kafka/server/ConfigHandler.scala

```diff
@@ -55,15 +56,27 @@ class TopicConfigHandler(private val logManager: LogManager) extends ConfigHandl
   }
 }
 
+object ClientConfigOverride {
+  val ProducerOverride = "producer_byte_rate"
+  val ConsumerOverride = "consumer_byte_rate"
+}
+
 /**
  * The ClientIdConfigHandler will process clientId config changes in ZK.
  * The callback provides the clientId and the full properties set read from ZK.
- * This implementation does nothing currently. In the future, it will change quotas per client
+ * This implementation reports the overrides to the respective ClientQuotaManager objects
  */
-class ClientIdConfigHandler extends ConfigHandler {
-  val configPool = new Pool[String, Properties]()
+class ClientIdConfigHandler(private val quotaManagers: Map[Short, ClientQuotaManager]) extends ConfigHandler {
 
-  def processConfigChanges(clientId : String, clientConfig : Properties): Unit = {
-    configPool.put(clientId, clientConfig)
+  def processConfigChanges(clientId : String, clientConfig : Properties) = {
+    if (clientConfig.containsKey(ClientConfigOverride.ProducerOverride)) {
+      quotaManagers.get(RequestKeys.ProduceKey).get.updateQuota(clientId,
```

Comment:
This can just be quotaManagers(RequestKeys.ProduceKey).updateQuota(clientId .... Ditto below.